src/main/java/org/kainos/ea/db/AuthDAO.java

```diff
@@ -0,0 +1,76 @@
+package org.kainos.ea.db;
+
+import org.kainos.ea.cli.LoginDetails;
+import org.kainos.ea.cli.LoginRequest;
+import org.kainos.ea.client.DatabaseConnectionFailedException;
+import org.kainos.ea.client.InvalidLoginAttemptException;
+import org.kainos.ea.client.CouldNotFindUserAccountException;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class AuthDAO {
+
+  private final DatabaseConnector databaseConnector = new DatabaseConnector();
+
+  public LoginRequest getUserPasswordHash (LoginRequest loginRequest) throws
+          InvalidLoginAttemptException, DatabaseConnectionFailedException {
+
+    String selectStatement = "SELECT passwordHash " +
+            "FROM `Users` WHERE email = ?";
+    // Check User Login Details Are Correct.
+    try (Connection c = databaseConnector.getConnection();
+            PreparedStatement st = c.prepareStatement(selectStatement)) {
+
+      st.setString(1, loginRequest.getEmail());
+
+      ResultSet rs = st.executeQuery();
+
+      while (rs.next()) {
```

Comment:
This can be an `if` statement instead of a `while` loop. In theory you should only get 1 or 0 results.

Reply:
Makes sense. All fixed :)